pdp-server/src/api/authzen/evaluations.rs

```diff
@@ -0,0 +1,376 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use utoipa::ToSchema;
+
+/// AuthZen Evaluations Request - for batch evaluation of multiple access requests
+#[derive(Debug, Serialize, Deserialize, ToSchema, Clone, PartialEq)]
+pub struct AccessEvaluationsRequest {
+    /// Subject (user) making the request - used as default for all evaluations
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub subject: Option<AuthZenSubject>,
+
+    /// Resource being accessed - used as default for all evaluations
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub resource: Option<AuthZenResource>,
+
+    /// Action being performed - used as default for all evaluations
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub action: Option<AuthZenAction>,
+
+    /// Context for the evaluation - used as default for all evaluations
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub context: Option<HashMap<String, serde_json::Value>>,
+
+    /// List of individual evaluations to perform
+    pub evaluations: Vec<IndividualEvaluation>,
+
+    /// Options for controlling evaluation behavior
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub options: Option<HashMap<String, serde_json::Value>>,
+}
+
+/// Individual evaluation in a batch request
+#[derive(Debug, Serialize, Deserialize, ToSchema, Clone, PartialEq)]
+pub struct IndividualEvaluation {
+    /// Subject (user) making the request - overrides the default
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub subject: Option<AuthZenSubject>,
+
+    /// Resource being accessed - overrides the default
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub resource: Option<AuthZenResource>,
+
+    /// Action being performed - overrides the default
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub action: Option<AuthZenAction>,
+
+    /// Context for the evaluation - overrides the default
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub context: Option<HashMap<String, serde_json::Value>>,
+}
+
+/// AuthZen Evaluations Response - contains decisions for all requests
+#[derive(Debug, Serialize, Deserialize, ToSchema, Clone, PartialEq)]
+pub struct AccessEvaluationsResponse {
+    /// List of evaluation results
+    pub evaluations: Vec<EvaluationResult>,
+}
+
+/// Individual evaluation result
+#[derive(Debug, Serialize, Deserialize, ToSchema, Clone, PartialEq)]
+pub struct EvaluationResult {
+    /// The decision whether to allow or deny the action
+    pub decision: bool,
+
+    /// Optional additional context
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub context: Option<HashMap<String, serde_json::Value>>,
+}
+
+// Convert an IndividualEvaluation to AllowedQuery, using defaults when needed
+fn convert_to_allowed_query(
+    evaluation: &IndividualEvaluation,
+    defaults: &AccessEvaluationsRequest,
+) -> Option<AllowedQuery> {
```

Comment:
feels like this should be implemented as either an "into" or a "from"

Reply:
It's different, since this does some "merge" with the top-level defaults passed in the root object of the Evaluations API body